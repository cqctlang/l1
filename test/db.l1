
@define safe(fn) {
    applyk(fn,
            @lambda(r) { r; },
            @lambda(m) { printf("error: %s\n", m); },
            []);
}

printf("%s\n", @db{a});
printf("%s\n", @db {b});
printf("%s\n", @db {c
});
printf("%s\n", @db {
    d });
printf("%s\n", @db%{e%});
printf("%s\n", @db%{f%%%%%%%%%%%});
printf("%s\n", @db%%%%%%{g%%%%%%%%%%%});
printf("%s\n", @db abcde{abcdeabcde!abcde});

printf("%s\n", @db<a>);
printf("%s\n", @db[a]);
printf("%s\n", @db{a});
printf("%s\n", @db(a));
printf("%s\n", @db((a));
printf("%s\n", @db([<{(a));
printf("%s\n", @db([<{(}>]a));
printf("%s\n", @db%(a%));
printf("%s\n", @db%[b%]);
printf("%s\n", @db%{c%});
printf("%s\n", @db%<d%>);
printf("%s\n", @db%(a)%));
printf("%s\n", @db%[b]%]);
printf("%s\n", @db%{c}%});
printf("%s\n", @db%<d>%>);

safe(@lambda() { parse("@db"); });
safe(@lambda() { parse("@db{"); });
safe(@lambda() { parse("@db{ "); });
safe(@lambda() { parse("@db%%%%%%%%%%%{ "); });
safe(@lambda() { parse("@db%%%%%%%%%%%{ }"); });
//line numbers
safe(@lambda() { parse("@db%%%%%%%%%%%{\n\n}"); });
safe(@lambda() { parse("@db\n\n"); });
safe(@lambda() { parse("@db\n\n\n{"); });
