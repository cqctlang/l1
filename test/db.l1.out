a
b
c


    d 
e
f%%%%%%%%%%
g%%%%%
abcdeabcde!
a
a
a
a
(a
[<{(a
[<{(}>]a
a
b
c
d
a)
b]
c}
d>
error: (stdin):1:0: unexpected EOF
error: (stdin):1:0: could not find close of @db
error: (stdin):1:1: could not find close of @db
error: (stdin):1:11: could not find close of @db
error: (stdin):1:12: could not find close of @db
error: (stdin):3:1: could not find close of @db
error: (stdin):3:0: unexpected EOF
error: (stdin):4:0: could not find close of @db
